pkg/providers/aws/provider_redis_test.go

```diff
@@ -489,7 +549,13 @@ func TestAWSRedisProvider_deleteRedisCluster(t *testing.T) {
 				Logger:            testLogger,
 				CredentialManager: &CredentialManagerMock{},
 				ConfigManager:     &ConfigManagerMock{},
-				CacheSvc:          &mockElasticacheClient{replicationGroups: buildReplicationGroupReady()},
+				CacheSvc: buildMockElasticacheClient(func(elasticacheClient *mockElasticacheClient) {
+					elasticacheClient.describeReplicationGroupsFn = func(*elasticache.DescribeReplicationGroupsInput) (*elasticache.DescribeReplicationGroupsOutput, error) {
+						return &elasticache.DescribeReplicationGroupsOutput{
+							ReplicationGroups: buildReplicationGroupReady(),
```

Comment:
Same as above we can refactor these ReplicationGroup functions into a single function with an override.